src/azure-cli/azure/cli/command_modules/appconfig/_kv_import_helpers.py

```diff
@@ -0,0 +1,958 @@
+                )
+
+    except (UnicodeDecodeError, json.JSONDecodeError):
+        if detected_encoding == default_encoding:
+            raise
+
+        with io.open(file_path, "r", encoding=detected_encoding) as config_file:
+            if format_ == "json":
+                config_data = json.load(config_file)
+
+            elif format_ == "yaml":
+                for yaml_data in list(yaml.safe_load_all(config_file)):
+                    config_data.update(yaml_data)
+
+            elif format_ == "properties":
+                config_data = javaproperties.load(config_file)
+                logger.debug(
+                    "Importing feature flags from a properties file is not supported. If properties file contains feature flags, they will be imported as regular key-values."
+                )
+
+    return config_data
+
+
+def __read_features_from_file(file_path, format_):
+    config_data = {}
+    features_dict = {}
+
+    if format_ == "properties":
+        logger.warning(
+            "Importing feature flags from a properties file is not supported. If properties file contains feature flags, they will be imported as regular key-values."
+        )
+        return features_dict
+
+    try:
+        config_data = __read_with_appropriate_encoding(file_path, format_)
+        found_dotnet_schema = False
+        found_ms_fm_schema = False
+        dotnet_schema_keyword = None
+        for keywordset in (
+            FeatureManagementReservedKeywords.PASCAL,
+            FeatureManagementReservedKeywords.CAMEL,
+            FeatureManagementReservedKeywords.HYPHEN,
+        ):
+            # find the occurrences of feature management section in file.
+            if keywordset.feature_management in config_data:
+                if found_dotnet_schema:
+                    raise FileOperationError(
+                        'Unable to proceed because file contains multiple sections corresponding to "Feature Management".'
+                    )
+                features_dict[keywordset.feature_management] = config_data[
+                    keywordset.feature_management
+                ]
+                dotnet_schema_keyword = keywordset.feature_management
+                del config_data[keywordset.feature_management]
+                found_dotnet_schema = True
+
+        ms_feature_management_keyword = (
+            FeatureManagementReservedKeywords.UNDERSCORE.feature_management
+        )
+        if ms_feature_management_keyword in config_data:
+            if found_dotnet_schema and any(
+                key != FeatureFlagConstants.FEATURE_FLAGS_KEY
+                for key in config_data[ms_feature_management_keyword].keys()
+            ):
+                raise FileOperationError(
+                    "Data contains an already defined section with the key %s."
+                    % (dotnet_schema_keyword)
+                )
```

Comment:
If I understood correctly, this condition ensures that if dotnet schema exists in file AND "feature_management" section is also present, that section cannot contain any key other than "feature_flags".
Why don't we just ignore the sections that are not "feature_flags"?

Reply:
Yes if dotnet schema exists and "feature_management" section is also present and it contains other keys other than "feature_flags" this means that there is another dotnet schema already defined within the "feature_management" section. The idea is we don't want the dotnet schema defined twice so we throw because it's an invalid schema.

Reply:
We only support these scenarios:
```
featureManagement: {
//dotnet schema
},
feature_management: {
    feature_flags: [   //new schema]
}
```

```
FeatureManagement: {
//dotnet schema
},
feature_management: {
    feature_flags: [   //new schema]
}
```
```
feature-management: {
//dotnet schema
},
feature_management: {
    feature_flags: [   //new schema]
}
```

```
feature_management: {
    feature_flags: [
   //new schema
    ],
  //dotnet schema
}
```

Reply:
I see. So this condition is checking the last scenario from your example. Thanks!